src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -148,7 +148,8 @@ else if (fieldObject != null && fieldObject instanceof IJsonBackedObject) {
                     setChildAdditionalData((IJsonBackedObject) fieldObject,rawJson.get(field.getName()).getAsJsonObject());
                 }
             } catch (IllegalArgumentException | IllegalAccessException e) {
-                logger.logError("Unable to access child fields of " + serializedObject.getClass().getSimpleName(), e);
+                //Not throwing the IllegalArgumentException as the Serialized Object would still be usable even if the additional data is not set.
```

Comment:
I would argue that just because the Java reflection engine hands us a field that it won't let us access, isn't a good enough reason to throw an exception at the user.  Especially when the only reason we are reflecting over it is because we are hunting for extra properties that might be there so that we can fill the additionalProperties collection.  

I do agree that the try/catch is a bit too wide of a net and we should try and limit the lines we are actually swallowing exceptions for.

However, this brings me to my main concern here.  If I understand this code, we are reflecting over every single property of every object that is returned from the Graph.  This seems like an expensive solution to a problem that by definition is an exception case.  The .Net library has the luxury of doing this because the deserializer library builds the list of unknown properties while it is parsing the JSON.  

I really don't like the idea of running this chunk of reflection code on the retrieval of every single object.  That just sounds like asking for trouble.

Reply:
@davidmotel,I agree that 100% code coverage is the ideal test case result that we should strive for. And we will further try to enhance this test case to cover most of the branching in the code.
As for the exceptions caught at the end of setting up additional data, these exceptions are thrown by field.get(object) method of  java.lang.reflect.Field class. So, we are just catching it and not throwing it to break the flow.

Reply:
No, you are catching IllegalArgumentException and IllegalAccessException. A NullPointerException will be thrown not caught. This is why unit tests are important.